gneiss/plot/_dendrogram.py

```diff
@@ -41,9 +46,34 @@ def __init__(self, use_lengths=True, **kwargs):
     def _cache_ntips(self):
         for n in self.postorder():
             if n.is_tip():
-                n._n_tips = 1
+                n.leafcount = 1
+            else:
+                n.leafcount = sum(c.leafcount for c in n.children)
+
+    def update_geometry(self, use_lengths, depth=None):
+        """Calculate tree node attributes such as height and depth.
+        Despite the name this first pass is ignorant of issues like
+        scale and orientation"""
+        if self.length is None or not use_lengths:
+            if depth is None:
+                self.length = 0
             else:
-                n._n_tips = sum(c._n_tips for c in n.children)
+                self.length = 1
+        else:
+            self.length = self.length
+
+        self.depth = (depth or 0) + self.length
+
+        children = self.children
+        if children:
+            for c in children:
+                c.update_geometry(use_lengths, self.depth)
+            self.height = max([c.height for c in children]) + self.length
```

Comment:
`height` is the total height of the subtree.  `length` is the branch length connecting to that subtree.